test/integration/lib/update-lock.spec.ts

```diff
@@ -281,4 +284,117 @@ describe('lib/update-lock', () => {
 			);
 		});
 	});
+
+	describe('composition step actions', () => {
+		const lockdir = pathOnRoot(updateLock.BASE_LOCK_DIR);
+		const serviceLockPaths = {
+			1: [
+				path.join(lockdir, '1', 'server', 'updates.lock'),
+				path.join(lockdir, '1', 'server', 'resin-updates.lock'),
+				path.join(lockdir, '1', 'client', 'updates.lock'),
+				path.join(lockdir, '1', 'client', 'resin-updates.lock'),
+			],
+			2: [
+				path.join(lockdir, '2', 'main', 'updates.lock'),
+				path.join(lockdir, '2', 'main', 'resin-updates.lock'),
+			],
+		};
+
+		describe('takeLock', () => {
+			// TODO
+		});
+
+		describe('releaseLock', () => {
+			let testFs: TestFs.Enabled;
+
+			beforeEach(async () => {
+				testFs = await testfs(
+					{},
+					{ cleanup: [path.join(lockdir, '*', '*', '**.lock')] },
+				).enable();
+				// TODO: Update mocha-pod to work with creating empty directories
```

Comment:
Did you try using `{}` to indicate an empty dir? e.g.

```
testfs({'/tmp/balena/myappdir': {}})
```

Reply:
I commented about this in another comment chain, but this didn't work consistently for me :( will be sure to fix in a future PR